LearnositySDK/Request/DataApi.cs

```diff
@@ -160,7 +160,7 @@ private JsonObject handleRequestRecursive(string url, JsonObject securityPacket,
                     }
                 }
             }
-            while(Tools.array_key_exists("next", meta));
+            while(Tools.array_key_exists("next", meta) && !Tools.empty(data.getJsonObject("data")));
```

Comment:
@gonzalozawa hey, makes sense, I'd only suggest to do a small refactor to make it better aligned with PHP SDK: https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/LearnositySdk/Request/DataApi.php